src/engine/controls/bpmcontrol.cpp

```diff
@@ -470,6 +470,11 @@ double BpmControl::calcSyncAdjustment(bool userTweakingSync) {
     double syncTargetBeatDistance = m_dSyncTargetBeatDistance.getValue();
     // We want the untweaked beat distance, so we have to add the offset here.
     double thisBeatDistance = m_pThisBeatDistance->get() + m_dUserOffset.getValue();
```

Comment:
Below (line 496) we aim to undo the added offset. This will move in the opposite direction if we are in reverse mode.

Reply:
oo good point